src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.PackageValidation.targets

```diff
@@ -0,0 +1,45 @@
+    <PackageDownload Include="$([MSBuild]::ValueOrDefault('$(PackageValidationBaselineName)', '$(PackageId)'))" Version="[$(PackageValidationBaselineVersion)]" />
+  </ItemGroup>
+
+  <Target Name="RunPackageValidation"
+          AfterTargets="Pack"
+          Condition="'$(IsPackable)' == 'true' and '$(EnablePackageValidation)' == 'true'">
+
+    <PropertyGroup Condition="'$(RoslynAssemblyPath)' == ''">
+      <RoslynAssemblyPath>$(RoslynTargetsPath)</RoslynAssemblyPath>
+      <RoslynAssemblyPath Condition="'$(MSBuildRuntimeType)' == 'Core'">$([System.IO.Path]::Combine('$(RoslynAssemblyPath)', bincore))</RoslynAssemblyPath>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <PackageValidationBaselineName Condition="'$(PackageValidationBaselineName)' == ''">$(PackageId)</PackageValidationBaselineName>
+      <PackageValidationBaselinePath Condition="'$(PackageValidationBaselinePath)' == '' and '$(PackageValidationBaselineVersion)' != ''">$([MSBuild]::NormalizePath('$(NuGetPackageRoot)', '$(PackageValidationBaselineName.ToLower())', '$(PackageValidationBaselineVersion)', '$(PackageValidationBaselineName.ToLower()).$(PackageValidationBaselineVersion).nupkg'))</PackageValidationBaselinePath>
+      <GenerateCompatibilitySuppressionFile Condition="'$(GenerateCompatibilitySuppressionFile)' == ''">false</GenerateCompatibilitySuppressionFile>
+      <_compatibilitySuppressionFilePath>$([MSBuild]::NormalizePath('$(MSBuildProjectDirectory)', 'CompatibilitySuppressions.xml'))</_compatibilitySuppressionFilePath>
+      <CompatibilitySuppressionFilePath Condition="'$(CompatibilitySuppressionFilePath)' == '' and (Exists($(_compatibilitySuppressionFilePath)) or '$(GenerateCompatibilitySuppressionFile)' == 'true')">$(_compatibilitySuppressionFilePath)</CompatibilitySuppressionFilePath>
+    </PropertyGroup>
+
+    <!-- PackageTargetPath isn't exposed by NuGet: https://github.com/NuGet/Home/issues/6671. -->
+    <Microsoft.DotNet.PackageValidation.ValidatePackage
+      PackageTargetPath="$([MSBuild]::ValueOrDefault('$(PackageTargetPath)', '$([MSBuild]::NormalizePath('$(PackageOutputPath)', '$(PackageId).$(PackageVersion).nupkg'))'))"
+      RuntimeGraph="$(RuntimeIdentifierGraphPath)"
+      NoWarn="$(NoWarn)"
+      RunApiCompat="$([MSBuild]::ValueOrDefault('$(RunApiCompat)', 'true'))"
+      EnableStrictModeForCompatibleTfms="$([MSBuild]::ValueOrDefault('$(EnableStrictModeForCompatibleTfms)', 'true'))"
+      EnableStrictModeForCompatibleFrameworksInPackage="$([MSBuild]::ValueOrDefault('$(EnableStrictModeForCompatibleFrameworksInPackage)', 'false'))"
+      GenerateCompatibilitySuppressionFile="$(GenerateCompatibilitySuppressionFile)"
+      CompatibilitySuppressionFilePath="$(CompatibilitySuppressionFilePath)"
+      BaselinePackageTargetPath="$(PackageValidationBaselinePath)"
+      DisablePackageBaselineValidation="$(DisablePackageBaselineValidation)"
+      RoslynAssemblyPath="$(RoslynAssemblyPath)"
+      RoslynMinVersion="$(RoslynMinVersion)" />
```

Comment:
Please remove this parameter. It is not an external decision. This is determined by what we compile against and should be observed by the version requested in the assembly resolve handler.